src/Microsoft.CodeAnalysis.Analyzers/UnitTests/MetaAnalyzers/MissingGeneratorAttributeRuleTests.cs

```diff
@@ -0,0 +1,406 @@
+﻿// Copyright (c) Microsoft.  All Rights Reserved.  Licensed under the Apache License, Version 2.0.  See License.txt in the project root for license information.
+
+using System.Threading.Tasks;
+using Microsoft.CodeAnalysis.Testing;
+using Test.Utilities;
+using Xunit;
+using VerifyCS = Test.Utilities.CSharpCodeFixVerifier<
+    Microsoft.CodeAnalysis.Analyzers.MetaAnalyzers.SourceGeneratorAttributeAnalyzer,
+    Microsoft.CodeAnalysis.Analyzers.MetaAnalyzers.Fixers.SourceGeneratorAttributeAnalyzerFix>;
+
+using VerifyVB = Test.Utilities.VisualBasicCodeFixVerifier<
+    Microsoft.CodeAnalysis.Analyzers.MetaAnalyzers.SourceGeneratorAttributeAnalyzer,
+    Microsoft.CodeAnalysis.Analyzers.MetaAnalyzers.Fixers.SourceGeneratorAttributeAnalyzerFix>;
+
+namespace Microsoft.CodeAnalysis.Analyzers.UnitTests.MetaAnalyzers
+{
+    public class MissingGeneratorAttributeRuleTests
+    {
+        private const string SourceGeneratorStub_CSharp = @"
+using System;
+using Microsoft.CodeAnalysis;
+
+namespace Microsoft.CodeAnalysis
+{
+    [AttributeUsage(AttributeTargets.Class)]
+    public sealed class GeneratorAttribute : Attribute
+    {
+    }
+
+    public interface ISourceGenerator
+    {
+        void Initialize(GeneratorInitializationContext context);
+        void Execute(GeneratorExecutionContext context);
+    }
+
+    public struct GeneratorInitializationContext
+    {
+    }
+
+    public readonly struct GeneratorExecutionContext
+    {
+    }
+}";
+
+        private const string SourceGeneratorStub_VisualBasic = @"
+Imports System
+Imports Microsoft.CodeAnalysis
+
+Namespace Microsoft.CodeAnalysis
+    <AttributeUsage(AttributeTargets.Class)>
+    public Class GeneratorAttribute
+            Inherits Attribute
+    End Class
+
+    public interface ISourceGenerator
+        Sub Initialize(context As GeneratorInitializationContext)
+        Sub Execute(context As GeneratorExecutionContext)
+    End Interface
+
+    public Class GeneratorInitializationContext
+    End Class
+
+    public Class GeneratorExecutionContext
+    End Class
+End Namespace";
+
+        [Fact]
+        public async Task TestMissing_CSharp()
+        {
+            var code = @"
+using Microsoft.CodeAnalysis;
+using Microsoft.CodeAnalysis.Diagnostics;
+
+[Generator]
+public class CustomGenerator : ISourceGenerator
+{
+    public void Initialize(GeneratorInitializationContext context) {}
+
+    public void Execute(GeneratorExecutionContext context)
+    {
+    }
+}";
+
+            await new VerifyCS.Test
+            {
+                TestState =
+                {
+                    Sources = { code , SourceGeneratorStub_CSharp },
+                    AdditionalReferences = { AdditionalMetadataReferences.CodeAnalysisReference}
```

Comment:
I doubt this is needed since you're using those stubs.

Reply:
📝 As part of https://github.com/dotnet/roslyn-analyzers/pull/5235/files#r668021528, you'll want to remove `AdditionalReferences` anyway. `ReferenceAssemblies` is preferred for all cases where the definition comes from another pipeline (e.g. tests in dotnet/roslyn can use `AdditionalReferences`, but tests in dotnet/roslyn-analyzers would use `ReferenceAssemblies`).

Reply:
Good to know. Removed